docs/src/extend/custom-rules.md

```diff
@@ -542,76 +569,102 @@ module.exports = {
-* `text` - the full text of the code being linted. Unicode BOM has been stripped from this text.
-* `ast` - the `Program` node of the AST for the code being linted.
-* `scopeManager` - the [ScopeManager](./scope-manager-interface#scopemanager-interface) object of the code.
-* `visitorKeys` - the visitor keys to traverse this AST.
-* `lines` - an array of lines, split according to the specification's definition of line breaks.
+* `hasBOM`: (`boolean`) The flag to indicate whether the source code has Unicode BOM.
+* `text`: (`string`) The full text of the code being linted. Unicode BOM has been stripped from this text.
+* `ast`: (`object`) `Program` node of the AST for the code being linted.
+* `scopeManager`: [ScopeManager](./scope-manager-interface#scopemanager-interface) object of the code.
+* `visitorKeys`: (`object`) Visitor keys to traverse this AST.
+* `lines`: (`array`) Array of lines, split according to the specification's definition of line breaks.
 
 You should use a `SourceCode` object whenever you need to get more information about the code being linted.
 
-#### Deprecated
+#### Accessing the Source Text
 
-Please note that the following methods have been deprecated and will be removed in a future version of ESLint:
+If your rule needs to get the actual JavaScript source to work with, then use the `sourceCode.getText()` method. This method works as follows:
 
-* `getComments()` - replaced by `getCommentsBefore()`, `getCommentsAfter()`, and `getCommentsInside()`
-* `getTokenOrCommentBefore()` - replaced by `getTokenBefore()` with the `{ includeComments: true }` option
-* `getTokenOrCommentAfter()` - replaced by `getTokenAfter()` with the `{ includeComments: true }` option
-* `isSpaceBetweenTokens()` - replaced by `isSpaceBetween()`
-* `getJSDocComment()`
+```js
+
+// get all source
+var source = sourceCode.getText();
+
+// get source for just this AST node
+var nodeSource = sourceCode.getText(node);
+
+// get source for AST node plus previous two characters
+var nodeSourceWithPrev = sourceCode.getText(node, 2);
+
+// get source for AST node plus following two characters
+var nodeSourceWithFollowing = sourceCode.getText(node, 0, 2);
+```
+
+In this way, you can look for patterns in the JavaScript text itself when the AST isn't providing the appropriate data (such as the location of commas, semicolons, parentheses, etc.).
+
+#### Accessing Comments
+
+While comments are not technically part of the AST, ESLint provides a few ways for rules to access them:
+
+* `sourceCode.getAllComments()`: Return an array of all the comments found in the program. This is useful for rules that need to check all comments regardless of location.
+* `sourceCode.getCommentsBefore()`: Return an array of comments that appear directly before nodes.
+* `sourceCode.getCommentsAfter()`: Return an array of comments that appear directly after nodes.
+* `sourceCode.getCommentsInside()` Return an array of comments that appear inside nodes.
```

Comment:
```suggestion
* `sourceCode.getAllComments()`: Returns an array of all the comments found in the program. This is useful for rules that need to check all comments regardless of location.
* `sourceCode.getCommentsBefore()`: Returns an array of comments that appear directly before nodes.
* `sourceCode.getCommentsAfter()`: Returns an array of comments that appear directly after nodes.
* `sourceCode.getCommentsInside()` Returns an array of comments that appear inside nodes.
```

Reply:
Actually, all these methods are already described in the parent section, so this list looks like a duplicate. I think this section should only provide a brief overview.

Reply:
Actually, all these methods are already described in the parent section, so this list looks like a duplicate. I think this section should only provide a brief overview.

Reply:
adding parens in parent section linking to this section.